libass/ass_fontconfig.c

```diff
@@ -109,23 +110,26 @@ static void scan_fonts(FcConfig *config, ASS_FontProvider *provider)
         // simple types
         result  = FcPatternGetInteger(pat, FC_SLANT, 0, &meta.slant);
         result |= FcPatternGetInteger(pat, FC_WIDTH, 0, &meta.width);
-        result |= FcPatternGetInteger(pat, FC_WEIGHT, 0, &weight);
+        result |= FcPatternGetDouble(pat, FC_WEIGHT, 0, &weight);
         result |= FcPatternGetInteger(pat, FC_INDEX, 0, &index);
         if (result != FcResultMatch)
             continue;
 
         // fontconfig uses its own weight scale, apparently derived
         // from typographical weight. we're using truetype weights, so
-        // convert appropriately
-#if FC_VERSION >= 21191
+        // convert appropriately.
+#if FC_VERSION >= 21292
+        meta.weight = FcWeightToOpenTypeDouble(weight);
+#elif FC_VERSION >= 21191
+        // Versions prior to 2.12.92 only had integer precision.
         meta.weight = FcWeightToOpenType(weight);
```

Comment:
Should this round the now-fractional `weight` rather than truncate it? Or does old Fontconfig always produce an integer `weight` despite the `FcPatternGetDouble` call?

Reply:
Hmm, I suppose it's possible for an older fontconfig to give a fractional weight from `FcPatternGetDouble` if you upgraded, cached, then downgraded? (I don't think the cache version changed here). Seems pretty corner-case-y but I guess it doesn't hurt. Prior to the introduction of `FcWeightToOpenTypeDouble` and `FcWeightFromOpenTypeDouble`, the cached values were produced by the integer `FcWeightFromOpenType` function, so you couldn't end up with fractional weights.

As for rounding the return value from `FcWeightToOpenTypeDouble`, I suppose that wouldn't hurt either?